pkg/testing/v1/service.go

```diff
@@ -232,6 +232,22 @@ func WithServiceAnnotations(annotations map[string]string) ServiceOption {
 	}
 }
 
+// WithConfigAnnotations assigns config annotations to a service
+func WithConfigAnnotations(annotations map[string]string) ServiceOption {
+	return func(service *v1.Service) {
+		service.Spec.ConfigurationSpec.Template.ObjectMeta.Annotations = kmeta.UnionMaps(
+			service.Spec.ConfigurationSpec.Template.ObjectMeta.Annotations, annotations)
+
+	}
+}
+
+// WithConfigName sets the given name to the config spec
+func WithConfigName(name string) ServiceOption {
```

Comment:
We're actually setting the name on the revision template

```suggestion
// WithBYORevisionName sets the given name to the config spec
func WithBYORevisionName(name string) ServiceOption {
```